src/vec_int64_ppc.h

```diff
@@ -92,14 +100,752 @@
+ * \code
+static inline
+vb64_t
+vec_cmpequd (vui64_t a, vui64_t b)
+{
+  vb64_t result;
+#ifdef _ARCH_PWR8
+#if __GNUC__ >= 6
+  result = vec_cmpeq(a, b);
+#else
+  __asm__(
+      "vcmpequd %0,%1,%2;\n"
+      : "=v" (result)
+      : "v" (a),
+      "v" (b)
+      : );
+#endif
+#else
+  // _ARCH_PWR7 implementation ...
+#endif
+  return (result);
+}
+ * \endcode
+ * The implementation checks of the compile target is POWER8 then
+ * checks of the compiler is new enough to use the generic vector
+ * compare built-in. If the generic built-in is not defined in
+ * <altivec.h> then we provide the equivalent inline assembler.
+ *
+ * For POWER7 targets we don't have any vector compare doubleword
+ * operations and we need to define the equivalent operation using
+ * PowerISA 2.06B (and earlier) instructions.
+ * For example:
+ * \code
+#else
+  // _ARCH_PWR7 implementation ...
+  vui8_t permute =
+    { 0x04,0x05,0x06,0x07, 0x00,0x01,0x02,0x03,
+      0x0C,0x0D,0x0E,0x0F, 0x08,0x09,0x0A,0x0B};
+  vui32_t r, rr;
+  r = (vui32_t)vec_cmpeq((vui32_t)a, (vui32_t)b);
+  if (vec_any_ne((vui32_t)a, (vui32_t)b))
+    {
+       rr = vec_perm (r, r, permute);
+       r= vec_and (r, rr);
+    }
+  result = (vb64_t)r;
+#endif
+ * \endcode
+ * Here we use compare equal unsigned word. If all words are equal
+ * use the result as is.  Otherwise, if any word elements are not
+ * equal we do some extra work. For each doubleword, rotate the word
+ * compare result by 32-bits (here we use permute as we don't
+ * have rotate doubleword either). Then logical and the original word
+ * compare and rotated results to get the final doubleword compare
+ * results.
+ *
+ * Similarly for all the doubleword compare variants. Similarly for
+ * doubleword; add, subtract, maximum, minimum, shift, rotate,
+ * count leading zeros, population count, and Byte reverse.
+ *
+ * \subsection i64_missing_ops_0_2 More Challenges
+ * Now we can look at the case where vector doubleword operations,
+ * of interest, don't have an equivalent instruction.
+ * Here interesting operations include those that are supported
+ * for other element sizes and types.
+ *
+ * The simplest example is absolute difference which was introduced in
+ * PowerISA 3.0 for byte, halfword and word elements.
+ * From the implementation of vec_absduw() we see how to implement the
+ * operation for POWER8 using subtract, maximum, and minimum.
+ * For example:
+ * \code
+static inline vui64_t
+vec_absdud (vui64_t vra, vui64_t vrb)
+{
+  return vec_subudm (vec_maxud (vra, vrb), vec_minud (vra, vrb));
+}
+ * \endcode
+ * This works because pveclib provides implementations for min, max,
+ * and sub operations that work across GCC versions and provide
+ * processor specific implementations for POWER8/9 and POWER7.
+ *
+ * Now we need to look at the multiply doubleword situation. We need
+ * implementations for vec_msumudm(), vec_muleud(), vec_mulhud(),
+ * vec_muloud(), and vec_muludm(). As we saw in the implementations of
```

Comment:
s/As we saw/We saw/ to match "that" later.

Reply:
OK